SharpRepository.EfRepository/EfRepositoryBase.cs

```diff
@@ -14,11 +14,12 @@ public class EfRepositoryBase<T, TKey> : LinqRepositoryBase<T, TKey> where T : c
     {
         protected IDbSet<T> DbSet { get; private set; }
         protected DbContext Context { get; private set; }
+        public bool GenerateKeyOnAdd { get; set; }
```

Comment:
Feels like this should be further down in the RepositoryBase and IRepository so it's not dependent on knowing you have an EfRepository or an InMemoryRepository instance